packages/abi-utils/lib/arbitrary.ts

```diff
@@ -231,124 +231,147 @@ const Type: fc.Memo<string> = fc.memo(n =>
       fc.oneof(Primitive(), ArrayFixed(n > 3 ? 3 : n), ArrayDynamic(n))
 );
 
-const ArrayFixed = fc.memo(n => {
-  const tuple =
-    n <= 1
-      ? fc.tuple(Primitive(), fc.integer(1, 256))
-      : fc.tuple(Type(), fc.integer(1, 256));
-
-  return tuple.map(([type, length]) => `${type}[${length}]`);
-});
-
-const ArrayDynamic = fc.memo(n => {
-  const item = n <= 1 ? Primitive() : Type();
+const ArrayFixed = fc.memo(n =>
+  fc
+    .tuple(Type(n - 1), fc.integer(1, 256))
```

Comment:
Note, my understanding is that the explicit `n-1` is unnecessary, but I haven't tried it and don't quite understand how this works, so IDK.  Anyway being more explicit isn't a bad thing, so, whatever.

Reply:
Yeah I decided to be more explicit because it seemed appropriate here.

Reply:
Cause otherwise `n` would be unused and such, and then someone might ask "why do you need `fc.memo`?"... and yeah